Client/Frontend/Settings/AppSettingsTableViewController.swift

```diff
@@ -80,6 +90,31 @@ class AppSettingsTableViewController: SettingsTableViewController {
             accountSectionTitle = NSAttributedString(string: Strings.FxAFirefoxAccount)
         }
 
+        let engineSettingChanged: ((String) -> ((Bool)->Void)) = { engineName in
+            let prefName = "sync.engine.\(engineName).enabledStateChanged"
+            return { enabled in
+                if let _ = prefs.boolForKey(prefName) { // Switch it back to not-changed
+                    prefs.removeObjectForKey(prefName)
+                } else {
+                    prefs.setBool(true, forKey: prefName)
+                }
+                if enabled {
+                    self.enginesToDisable.remove(engineName)
+                    _ = self.profile.syncManager.syncNamedCollections(why: SyncReason.engineEnabled, names: [engineName])
```

Comment:
Question: what happens if I tap-tap on Logins? Do we spend 30 seconds uploading ours to the server, then delete them when we leave the settings screen? Is there a better option?

Reply:
The tap-tap story is not great in my opinion, I believe we should start syncing (disablements AND enablements) after the user leaves the screen. This will avoid a lot of headaches and UI/sync races.